libsolidity/ast/Types.cpp

```diff
@@ -339,29 +361,159 @@ MemberList::MemberMap IntegerType::nativeMembers(ContractDefinition const*) cons
+		dev::toString(_Lbits) + "x" + dev::toString(_Rbits)
+	);
+}
+
+bool FixedType::isImplicitlyConvertibleTo(Type const& _convertTo) const
+{
+	if (_convertTo.category() == category())
+	{
+		FixedType const& convertTo = dynamic_cast<FixedType const&>(_convertTo);
+		if (convertTo.m_leftBits < m_leftBits || convertTo.m_rightBits < m_rightBits)
+			return false;
+		else if (isSigned())
+			return convertTo.isSigned();
+		else
+			return !convertTo.isSigned() || (convertTo.m_leftBits > m_leftBits && convertTo.m_rightBits > m_rightBits);
+	}
+	else if (_convertTo.category() == Category::Integer)
+	{
+		IntegerType const& convertTo = dynamic_cast<IntegerType const&>(_convertTo);
+		if (convertTo.numBits() < m_leftBits)
+			return false;
+		else if (isSigned())
+			return convertTo.isSigned();
+		else
+			return !convertTo.isSigned() || convertTo.numBits() > m_leftBits;
+	}
+	else
+		return false;
+}
+
+bool FixedType::isExplicitlyConvertibleTo(Type const& _convertTo) const
+{
+	return _convertTo.category() == category() ||
+		_convertTo.category() == Category::Integer ||
+		_convertTo.category() == Category::Enum ||
+		_convertTo.category() == Category::FixedBytes;
+}
+
+TypePointer FixedType::unaryOperatorResult(Token::Value _operator) const
+{
+	// "delete" is ok for all fixed types
+	if (_operator == Token::Delete)
+		return make_shared<TupleType>();
+	// for fixed, we allow +, -, ++ and --
+	else if (
+		_operator == Token::Add || 
+		_operator == Token::Sub ||
+		_operator == Token::Inc || 
+		_operator == Token::Dec ||
+		_operator == Token::After
+	)
+		return shared_from_this();
+	else
+		return TypePointer();
+}
+
+bool FixedType::operator==(Type const& _other) const
+{
+	if (_other.category() != category())
+		return false;
+	FixedType const& other = dynamic_cast<FixedType const&>(_other);
+	return other.m_leftBits == m_leftBits && other.m_rightBits == m_rightBits && other.m_modifier == m_modifier;
+}
+
+string FixedType::toString(bool) const
+{
+	string prefix = isSigned() ? "fixed" : "ufixed";
+	return prefix + dev::toString(m_leftBits) + "x" + dev::toString(m_rightBits);
+}
+
+TypePointer FixedType::binaryOperatorResult(Token::Value _operator, TypePointer const& _other) const
+{
+	if (_other->category() != Category::NumberConstant 
+		&& _other->category() != category()
+		&& _other->category() != Category::Integer
+	)
+		return TypePointer();
+	auto commonType = dynamic_pointer_cast<FixedType const>(Type::commonType(shared_from_this(), _other));
+
+	if (!commonType)
+		return TypePointer();
+
+	// All fixed types can be compared
+	if (Token::isCompareOp(_operator))
+		return commonType;
+	if (Token::isBooleanOp(_operator))
+		return TypePointer();
+	return commonType;
+}
+
+bool ConstantNumberType::isValidLiteral(const Literal& _literal)
 {
 	try
 	{
-		bigint x(_literal.value());
+		string literal = _literal.value();
+		auto literalEnd = remove(literal.begin(), literal.end(), '.');
+		bigint x(string(literal.begin(), literalEnd));
```

Comment:
Definitely should have written that in. This checks to see if it is a valid literal or not. The remove function is to make sure that the radix is removed, otherwise there will be an error thrown when trying to convert to a bigint. Essentially what this is checking is that there are no alphabetical characters, and that there are no odd symbols in there that would not be considered part of an integer.